go.mod

```diff
@@ -0,0 +1,7 @@
+module istio.io/gogo-genproto
+
+go 1.12
+
+replace github.com/gogo/protobuf => github.com/istio/gogo-protobuf v1.2.2-0.20190726125433-4c9abdb3090c
+
+require github.com/gogo/protobuf v0.0.0-00010101000000-000000000000
```

Comment:
this gives me an error when I am trying to update dependencies:
```
go: finding github.com/gogo/protobuf v0.0.0-00010101000000-000000000000
go: github.com/gogo/protobuf@v0.0.0-00010101000000-000000000000: unknown revision 000000000000
go get: error loading module requirements
```

Reply:
Also getting this via something I depend on (and had to find my way here by grepping my module cache for this zero'd version).

I don't know anything about this repo, but it seems strange that this module is replacing another with itself inside of its own module. I wonder if this is exposing a Go bug where it's generating an impossible version and just hasn't happened until now. Might be good to report to see if this is expected behavior in Go 1.12. Things work fine at tip.

And unless I'm mistaken, this repo has no `main` package, so replacements here wouldn't have much of an effect, as users of this module will not see any replacements at all, but maybe it's being used to build the `pb.go` files and I'm missing it.

Reply:
I got the same error, also in Go 1.12:

```
go: github.com/gogo/protobuf@v0.0.0-00010101000000-000000000000: unknown revision 000000000000
go get: error loading module requirements
```

Reply:
Hi!

I have the same error.
`
go: finding github.com/envoyproxy/protoc-gen-validate v0.1.0
go: github.com/gogo/protobuf@v0.0.0-00010101000000-000000000000: unknown revision 000000000000
go get: error loading module requirements
`

I discovered this dependency chain:
MyProject -> google.golang.org/grpc -> github.com/envoyproxy/go-control-plane -> istio/gogo-genproto
So, I afraid that this bug affected many people

Reply:
I have temporary solution: write in your project go.mod
`
replace github.com/gogo/protobuf v0.0.0-00010101000000-000000000000 => github.com/istio/gogo-protobuf v1.2.2-0.20190726125433-4c9abdb3090c
`

Reply:
Sorry for the broken dependencies there.

We're trying to get  inconsistent versions of the gogo protobuf code synced up across repos in the Istio organization. We have our own fork of the gogo code (istio/gogo-protobuf)  which fixes a bug in that code base (while we wait for our PR to be merged into the original gogo repo).

So this repo, along with some others are being rebuilt with the fork as a dependency into of the original gogo code.

BTW, the reason for this repo is strictly to generate the .pb.go files which we use elsewhere  in Istio.

Reply:
>  which fixes a bug in that code base (while we wait for our PR to be merged into the original gogo repo).

@geeknoid which bug are you waiting on in gogo?

Reply:
This PR: https://github.com/gogo/protobuf/pull/569